addons/sale/models/payment_transaction.py

```diff
@@ -131,8 +131,23 @@ def _reconcile_after_done(self):
             ):
                 send_invoice_cron._trigger()
             else:
-                # Must be called after the super() call to make sure the invoice are correctly posted.
-                self._send_invoice()
+                self._send_invoice_post_commit()
+
+    def _send_invoice_post_commit(self):
+        if self.env['account.move']._can_commit():
+            payment_ids = self.ids
+            dbname = self.env.cr.dbname
+            context = self._context
```

Comment:
```suggestion
            context = self.env.context
```
`_context` will be deprecated soon